201-redis-vnet-geo-replication/azuredeploy.json

```diff
@@ -0,0 +1,147 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "The location of the existing primary virtual network and new primary cache."
+            }
+        },
+        "existingPrimaryVirtualNetworkName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing primary virtual network."
+            }
+        },
+        "existingPrimaryCacheSubnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing primary cache subnet."
+            }
+        },
+        "newPrimaryCacheName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the new primary cache."
+            }
+        },
+        "secondaryLocation": {
+            "type": "string",
+            "metadata": {
+                "description": "The location of the existing secondary virtual network and new secondary cache."
+            }
+        },
+        "existingSecondaryVirtualNetworkName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing secondary virtual network."
+            }
+        },
+        "existingSecondaryCacheSubnetName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the existing secondary cache subnet."
+            }
+        },
+        "newSecondaryCacheName": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the new secondary cache."
+            }
+        }
+    },
+    "variables": {
+        "primarySubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingPrimaryVirtualNetworkName'), parameters('existingPrimaryCacheSubnetName'))]",
```

Comment:
Since these are meant to be existing vnets, you need to add the resourceGroup name parameter to the function and will need to add to the parameters so it can be passed in.